app/src/main/java/com/mapbox/android/events/testapp/MainActivity.java

```diff
@@ -70,7 +72,32 @@ private void checkPermissions() {
 
     if (permissionsGranted) {
       mapboxTelemetry.enable();
-      locationEngine.requestLocationUpdates(getRequest(2000), callback, null);
+      LocationEngine locationEngine = LocationEngineProvider.getBestLocationEngine(this);
```

Comment:
locationEngine is a temporary variable, so it will be released out of this block and will not result in memory leak.

Reply:
It showcases a wrong setup which will result in leaks as @andrlee mentioned. Since we are aware of this, we should not include the example at all now and follow up in a separate PR.

Reply:
@LukasPaczos the right setup with other goodies just landed in master: https://github.com/mapbox/mapbox-events-android/pull/347